test/utils.js

```diff
@@ -9,10 +9,28 @@
  * OF ANY KIND, either express or implied. See the License for the specific language
  * governing permissions and limitations under the License.
  */
+const { fetch, timeoutSignal, AbortError } = require('@adobe/helix-fetch');
 const assert = require('assert');
 
 function assertContains(actual, patterns) {
   patterns.map((expected) => assert.ok(new RegExp(expected).test(actual), `${actual} does not match ${expected}`));
 }
 
-module.exports = { assertContains };
+async function fetchAndRetry(url, duration = 0, defaultTimeout = 300) {
+  if (duration >= 1000) {
```

Comment:
In the description of this PR you wrote:

> This code adds timeout at 3000ms with action failure; after retries at 500ms timeout intervals.

That's not what the code is doing. The timeout/retry interval is 300ms and the fetch fails after 1000ms.